### PR TITLE
doc: update spoon-smpl README and related docs

### PR DESCRIPTION
**Pull request by mkforsb**
Adds a table listing some of the currently supported SmPL features to the spoon-smpl README and fixes minor issues in related docs.

### CODE REVIEW

File: doc/spoon_smpl.md [hunk 1]
```diff
@@ -62,10 +62,10 @@ $ java -cp <classpath> spoon.smpl.CommandlineApplication
...
-3. [PatternBuilder.java](https://github.com/mkforsb/spoon/blob/smpl/spoon-smpl/src/main/java/spoon/smpl/pattern/PatternBuilder.java): shows the set of matchable Java elements, but can be cumbersome to read.
-4. [Substitutor.java](https://github.com/mkforsb/spoon/blob/smpl/spoon-smpl/src/main/java/spoon/smpl/Substitutor.java): shows the set of transformable Java elements, but can be cumbersome to read.
+1. [smpl_grammar.txt](https://github.com/INRIA/spoon/blob/master/spoon-smpl/smpl_grammar.txt): covers parts of the currently supported grammar.
+2. [Test cases](https://github.com/INRIA/spoon/tree/master/spoon-smpl/src/test/resources/endtoend): contains easy-to-read test cases that reveal both supported patch language features and supported Java elements.
+3. [PatternBuilder.java](https://github.com/INRIA/spoon/blob/master/spoon-smpl/src/main/java/spoon/smpl/pattern/PatternBuilder.java): shows the set of matchable Java elements, but can be cumbersome to read.
+4. [Substitutor.java](https://github.com/INRIA/spoon/blob/master/spoon-smpl/src/main/java/spoon/smpl/Substitutor.java): shows the set of transformable Java elements, but can be cumbersome to read.
```
Comment (mkforsb) [inline doc/spoon_smpl.md:68]: Update links to point to main INRIA repo.

File: spoon-smpl/README.md [hunk 2]
```diff
@@ -13,13 +13,40 @@ expression E;
...
+4. [Substitutor.java](https://github.com/INRIA/spoon/blob/master/spoon-smpl/src/main/java/spoon/smpl/Substitutor.java): shows the set of transformable Java elements, but can be cumbersome to read.
+
 Spoon-SmPL's initial version has been implemented by Mikael Forsberg as part of his master's thesis ["Design and Implementation of Semantic Patch Support for the Spoon Java Transformation Engine' at KTH Royal Institute of Technology"](http://urn.kb.se/resolve?urn=urn:nbn:se:kth:diva-291226).
 
 References:
 * [SmPL: A Domain-Specific Language for Specifying Collateral Evolutions in Linux Device Drivers](http://coccinelle.lip6.fr/papers/ercim.pdf)
 * [A Foundation for Flow-Based Program Matching Using Temporal Logic and Model Checking](http://coccinelle.lip6.fr/papers/popl09.pdf)
 * [Semantic Patches for Java Program Transformation](https://drops.dagstuhl.de/opus/volltexte/2019/10814/pdf/LIPIcs-ECOOP-2019-22.pdf)
-* [Semantic Patches for Java Program Transformation](https://drops.dagstuhl.de/opus/volltexte/2019/10814/pdf/LIPIcs-ECOOP-2019-22.pdf)
```
Comment (mkforsb) [inline spoon-smpl/README.md:22]: Remove duplicate paper reference.

File: spoon-smpl/smpl_grammar.txt [hunk 3]
```diff
@@ -23,7 +23,6 @@ TODO: markdownify this file
     metatype ::= identifier
                | type
                | expression
-               | statement
```
Comment (mkforsb) [inline spoon-smpl/smpl_grammar.txt:26]: Remove error.

**Comment by monperrus** [comment]
thanks @mkforsb

**Pull request merged**